core/src/mender-error-counters.c

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file      mender-error-counters.c
+ * @brief     Mender Error counters implementation
+ *
+ * Copyright Northern.tech AS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdint.h>
+
+#include "mender-error-counters.h"
+#include "mender-log.h"
+#include "mender-utils.h"
+
+#define NETWORK_ERR_LIMIT 30
+
+static uint8_t net_errors = 0;
+
+mender_err_t
+mender_err_count_net_inc(void) {
+    if (net_errors < UINT8_MAX) {
```

Comment:
This is okey as long as `#define NETWORK_ERR_LIMIT` is in this same file. But if we later move it around we risk having `NETWORK_ERR_LIMIT > 256` and then the error detection will never trigger.

(will the compiler throw a warning below for `net_errors > NETWORK_ERR_LIMIT` if NETWORK_ERR_LIMIT overflows `uint8_t`?)

Reply:
> will the compiler throw a warning below for `net_errors > NETWORK_ERR_LIMIT` if NETWORK_ERR_LIMIT overflows `uint8_t`

I don't think so. But we have multiple ways to make sure the limit is not above 255, right? Kconfig supports range, we can do `static_assert` or even 
```
#if NETWORK_ERR_LIMIT > UINT8_MAX
#error NETWORK_ERR_LIMIT cannot be greater than UINT8_MAX
#endif
```